src/network/networkctl.c

```diff
@@ -296,20 +297,26 @@ static int decode_link(sd_netlink_message *m, LinkInfo *info, char **patterns) {
 
         if (patterns) {
                 char str[DECIMAL_STR_MAX(int)];
+                size_t pos;
+
+                assert(matched_patterns);
 
                 xsprintf(str, "%i", ifindex);
-                if (!strv_fnmatch(patterns, str, 0) && !strv_fnmatch(patterns, name, 0)) {
+                if (!strv_fnmatch_full(patterns, str, 0, &pos) &&
+                    !strv_fnmatch_full(patterns, name, 0, &pos)) {
                         bool match = false;
                         char **p;
 
                         STRV_FOREACH(p, altnames)
-                                if (strv_fnmatch(patterns, *p, 0)) {
+                                if (strv_fnmatch_full(patterns, *p, 0, &pos)) {
                                         match = true;
                                         break;
                                 }
                         if (!match)
                                 return 0;
                 }
+
+                matched_patterns[pos] = true;
```

Comment:
note that this is not entirely correct, since if i have one interface "eth0" and I look for "eth*" and for "*0" then this code would complain that "*0" is never matched. Do we care?

Reply:
Just at debug level. I think that's OK.
```console
$ SYSTEMD_LOG_LEVEL=debug build/networkctl list --no-legend hub* *ub0
Pattern "*ub0" doesn't match any interface, ignoring
Successfully forked off '(pager)' as PID 139689.
Skipping PR_SET_MM, as we don't have privileges.
Pager executable is "less", options "FRSXMK", quit_on_interrupt: yes
 10 hub0 ether no-carrier unmanaged
$ build/networkctl list --no-legend hub* *ub0
 10 hub0 ether no-carrier unmanaged
```

Reply:
... if anyone ever complains about this, this code could be changed to check if none of the interfaces match the pattern, but I strongly doubt anyone would ever care.